src/Sfa.Tl.ResultsAndCertification.Web/Loader/AdminPostResultsLoader.cs

```diff
@@ -60,5 +66,84 @@ public Task<AdminAddCoreRommOutcomeViewModel> GetAdminAddPathwayRommOutcomeAsync
+         => model.Grades = GetAdminAddRommOutcomeChangeGradeCoreGrades(await GetAdminChangeResultGrades(LookupCategory.PathwayComponentGrade, model.Grade, true));
+
+        private List<LookupViewModel> GetAdminAddRommOutcomeChangeGradeCoreGrades(List<LookupViewModel> Grades)
+        {
+            return Grades.Where(t => !t.Code.Equals(Constants.PathwayComponentGradeQpendingResultCode, StringComparison.InvariantCultureIgnoreCase)
+            && !t.Code.Equals(Constants.PathwayComponentGradeXNoResultCode, StringComparison.InvariantCultureIgnoreCase)
+            && !t.Code.Equals(Constants.NotReceived, StringComparison.InvariantCultureIgnoreCase)).ToList();
+        }
+
+
+        public async Task<AdminAddRommOutcomeChangeGradeSpecialismViewModel> GetAdminAddRommOutcomeChangeGradeSpecialismAsync(int registrationPathwayId, int assessmentId)
+        {
+            var viewmodel = await GetAdminAddResultAsync<AdminAddRommOutcomeChangeGradeSpecialismViewModel>(registrationPathwayId, assessmentId, LookupCategory.SpecialismComponentGrade, false);
+            viewmodel.Grades = GetAdminAddRommOutcomeChangeGradeSpecialismGrades(viewmodel.Grades);
+            viewmodel.Grades.Remove(viewmodel.Grades.Where(t => t.Value == viewmodel.Grade).FirstOrDefault());
+            return viewmodel;
+        }
+
+        public async Task LoadAdminAddRommOutcomeChangeGradeSpecialismGrades(AdminAddRommOutcomeChangeGradeSpecialismViewModel model)
+        => model.Grades = GetAdminAddRommOutcomeChangeGradeSpecialismGrades(await GetAdminChangeResultGrades(LookupCategory.SpecialismComponentGrade, model.Grade, true));
+
+
+        private List<LookupViewModel> GetAdminAddRommOutcomeChangeGradeSpecialismGrades(List<LookupViewModel> Grades)
+        {
+            return Grades.Where(t => !t.Code.Equals(Constants.SpecialismComponentGradeQpendingResultCode, StringComparison.InvariantCultureIgnoreCase)
+            && !t.Code.Equals(Constants.SpecialismComponentGradeXNoResultCode, StringComparison.InvariantCultureIgnoreCase)
+            && !t.Code.Equals(Constants.NotReceived, StringComparison.InvariantCultureIgnoreCase)).ToList();
+        }
+
+        private async Task<List<LookupViewModel>> GetAdminAddResultGrades(LookupCategory lookupCategory)
+        {
+            IList<LookupData> grades = await _internalApiClient.GetLookupDataAsync(lookupCategory);
+            return _mapper.Map<List<LookupViewModel>>(grades);
+        }
+
+        private async Task<List<LookupViewModel>> GetAdminChangeResultGrades(LookupCategory lookupCategory, string grade, bool isRomm = false)
+        {
+            IList<LookupData> grades = await _internalApiClient.GetLookupDataAsync(lookupCategory);
+            grades.Remove(grades.Where(t => t.Value == grade).FirstOrDefault());
+            if (!isRomm) grades.Insert(grades.Count, new LookupData { Code = Constants.NotReceived, Value = Content.Result.ManageSpecialismResult.Option_Remove_Result });
```

Comment:
Why grade.Insert and no grade.Add? You're adding an item at the end of the list right?